client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
I do not quite understand your last comment. I think the purpose is to auto sync. See the issue you are fixing for reference.